components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesBookmarksStorageTest.kt

```diff
@@ -263,5 +276,108 @@ class PlacesBookmarksStorageTest {
         verify(writer, times(1)).deleteBookmarkNode(newSeparator.guid)
     }
 
+    @Test
+    fun `build and submit telemetry pings on success`() = runBlocking {
+        val pingServer = getMockWebServer()
```

Comment:
I think we don't really need to do this dance, see my comments above. The only requirement I see, is that we have the possibility of calling the testing API before the ping is actually sent. This could be achieved without touching the Glean internals, by slightly refactoring the implementation. What do you think, @linacambridge ?

Reply:
Thanks, that does simplify things a lot! Since `Connection` is an internal interface, anyway (though it wasn't before; hopefully, @grigoryk is fine with that), I added separate `assemble{History, Bookmarks}Ping` and `send{History, Bookmarks}Ping` methods, with default implementations. The `RustPlacesConnection` interface actually implements the latter. That felt a little easier than futzing around with Mockito spies for `PlacesApi` and `RustPlacesConnection`.